common/common.h

```diff
@@ -20,6 +20,11 @@
 #define DIRECTORY_SEPARATOR '/'
 #endif // _WIN32
 
+#define print_build_info() do {                                                             \
```

Comment:
I think you can just use `{}`, it's definitely valid c/cpp to use `{}` without a loop, unless you deliberately want it to be a loop

Reply:
The benefit of the do/while(0) pattern is that it requires the caller to put a semicolon after. `print_build_info();` looks more natural than `PRINT_BUILD_INFO` (no parens or semi).

Reply:
Oh, that's clever, I think I'm gonna steal this idea :)

Did you mark me as reviewer, or did it happen automatically when I added a code comment ? I've been using gitlab for past couple of years and I don't remember github specific quirks :)

Reply:
It happened automatically, but all it means is that you left a comment. You don't have to approve/reject my PR or anything.

Reply:
Ok.

Another thing I noticed, which might be a problem with rearranging includes, common adds another implementation of llama_token_to_piece, which may or may not cause some issues

I had a problem with it when I was looking for a place to include log.h, and that's the reason pretty print for tokens is a lambda, because that let function names be resolved in calling context, otherwise the other llama_token_to_piece was taking precedence, without fighting includes

Reply:
That shouldn't be an issue, because common.h is a C++ header, and its llama_token_to_piece takes only two arguments instead of four - so it's an overload. I think the only way that could cause a problem is if you forgot to include the appropriate header.